cmd/object-api-common.go

```diff
@@ -67,6 +69,12 @@ func dirObjectInfo(bucket, object string, size int64, metadata map[string]string
 	// return success.
 	md5Sum := metadata["md5Sum"]
 	delete(metadata, "md5Sum")
+	if md5Sum == "" {
```

Comment:
You could have a package-level constant for md5sum of empty payload instead of computing it every time this function is called.

```go
const emptyMD5Sum  = "d41d8cd98f00b204e9800998ecf8427e"
```